### PR TITLE
chapel 1.25.1

### PR DESCRIPTION
**Pull request by Maxrimus**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?
As a note, when I run `brew audit` I get `Dependency 'llvm@11' is an alias; use the canonical name 'llvm@11'.` which I don't know how to resolve

-----

### CODE REVIEW

**Comment by bradcray** [comment]
In case any reviewers happen to look at this:

* we believe the failing ubuntu checks are due to changes we made here and are looking into ways to address them

* on the 12-arm64 and 11-arm64 builds, the first errors we see are in the brew audit:
  ```
  chapel:
    * Binaries built for a non-native architecture were installed into chapel's prefix.
      The offending files are:
        /opt/homebrew/Cellar/chapel/1.25.1/libexec/third-party/qthread/qthread-src/installed/lib/libqthread.19.dylib	(x86_64)
      Unexpected universal binaries were found.
      The offending files are:
        /opt/homebrew/Cellar/chapel/1.25.1/libexec/third-party/chpl-venv/install/chpldeps/markupsafe/_speedups.cpython-39-darwin.so
  ```
  which we've [run into before](https://github.com/chapel-lang/chapel/issues/17910#issuecomment-950246280) when @carlocab's was helping us attempt to build arm bottles on homebrew.  We don't have any M1 Macs in house, and have not put effort into trying to work through those issues yet.  In our 1.25.0 formula,  it seems that this audit step was skipped:
  ```
  ==> Running Formulae#formula!(chapel)
==> SKIPPED chapel
Warning: chapel has not yet been bottled on ARM!
  ```
  It's not clear to me what would have caused the change in behavior here between the 1.25.0 formula and now.  Is it to others?
  
* later on, we see a failure to find gmp:
  ```
    ld: library not found for -lgmp
  clang: error: linker command failed with exit code 1 (use -v to see invocation)
  ```
  which is almost certainly due to changes in our formula to attempt to use homebrew GMP rather than our bundled version.  We're also looking into this but wanted to ask since the problem is specific to the arm configuration:  On M1 Macs, is there reason to believe that homebrew `gmp.h`/`libgmp.a` would be anywhere other than `/usr/local/[include|lib]`?

Thanks,
-Brad

**Comment by bradcray** [comment]
Commenting here for anyone subscribed to this PR to note that I've updated my previous comment to note that we've addressed the first bullet and think we have a lead on the third.

File: Formula/chapel.rb [hunk 1]
```diff
@@ -23,19 +23,16 @@ def install
     ENV["CHPL_HOME"] = libexec
     # This is for mason
     ENV["CHPL_RE2"] = "bundled"
+    ENV["CHPL_GMP"] = "system"
+    ENV["CHPL_INCLUDE_PATH"] = "/usr/local/include"
+    ENV["CHPL_LIB_PATH"] = "/usr/local/lib"
```
Comment (carlocab) [inline Formula/chapel.rb:28]:
```suggestion
    ENV["CHPL_INCLUDE_PATH"] = HOMEBREW_PREFIX/"include"
    ENV["CHPL_LIB_PATH"] = HOMEBREW_PREFIX/"lib"
```
Similarly for the test block.

File: Formula/chapel.rb [hunk 2]
```diff
@@ -23,19 +23,16 @@ def install
...
       system "./util/printchplenv", "--all"
       system "make"
-      # Need to let chapel choose target compiler with llvm
-      ENV["CHPL_HOST_CC"] = ENV["CC"]
-      ENV["CHPL_HOST_CXX"] = ENV["CXX"]
-      ENV.delete("CC")
-      ENV.delete("CXX")
-      system "./util/printchplenv", "--all"
-      system "make"
+      system "CHPL_LLVM=none make"
```
Comment (carlocab) [inline Formula/chapel.rb:35]:
```suggestion
      with_env(CHPL_LLVM: "none") do
        system "make"
      end
```
Similarly for the test block.

**Comment by bradcray** [comment]
Thanks for the feedback @carlocab .  The first change we were en route to making, but by running `brew --prefix` (where your approach is nicer).

Do you have any insights into why the arm audit is breaking our CI here, yet didn't for 1.25.0?  (since we know that 1.25.0 had the same problem)

**Review by chenrui333 — DISMISSED** [review]

**Comment by bradcray** [comment]
Hi @chenrui333 : Thanks for the approval.  We're not clear what to do about the failing audit checks on arm.  We know that [our code has problems on M1 Macs](https://github.com/chapel-lang/chapel/issues/17911), but we don't have access to an M1 Mac in order to reproduce and debug these issues (though we're working on getting one).  It looks as though in past releases (e.g., [Chapel 1.25.0](https://github.com/Homebrew/homebrew-core/pull/85813)), the arm audit step was somehow bypassed despite these issues.  Is there any way to proceed with this PR for the non-M1 and Ubuntu builds without also having full M1 support as well?  Thanks for any information.

File: Formula/chapel.rb [hunk 3]
```diff
@@ -60,8 +60,14 @@ def install
...
   test do
     ENV["CHPL_HOME"] = libexec
+    ENV["CHPL_GMP"] = "system"
+    ENV["CHPL_INCLUDE_PATH"] = HOMEBREW_PREFIX/"include"
+    ENV["CHPL_LIB_PATH"] = HOMEBREW_PREFIX/"lib"
     cd libexec do
       system "util/test/checkChplInstall"
+      with_env(CHPL_LLVM: "none") do
+        system "util/test/checkChplInstall"
+      end
```
Comment (mppf) [inline Formula/chapel.rb:70]: Can we repeat this also with `CHPL_TARGET_COMPILER: "system"` to make sure we really are getting an LLVM build and that the LLVM backend works?

**Comment by bradcray** [comment]
Closed in favor of #91257.

**Pull request closed without merging**